### PR TITLE
Element selected in Model Browser shows in Property Editor

### LINKED ISSUE #2545: Open element property page for an element selected from the model explorer
### Is your feature request related to a problem?
Sometimes I want to look/edit the details of an element without having to find and click on it from the diagram.

### Describe the solution you'd like
Clicking on an model element from the model explorer should open its property page.

### Describe alternatives you've considered
An alternative is to click on the model element from the diagram.

### Additional context
This feature would be first steps towards supporting modeling directly in the context menu, which is known to be practiced by more seasonal modelers because it enables to do some things much faster than through the diagram. A follow up that would allow to create some modeling elements using a context menu should follow to support that.

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Only items selected in a diagram are shown in the Property Editor.

Issue Number: Fixes #2545 

### What is the new behavior?

When you select an element in the Model Browser, it will show in the Property Editor.

Property pages that contain both model and presentation settings (normally a _Show Something_ setting) have been split in two pages.

### Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### Other information

Still to do:

- [x] Attributes and Operations for Block, Requirement, etc.
- [x] Select Extension relation in model browser causes error
- [x] ActivityParameterNode also shows ObjectNode properties

### CODE REVIEW

**Comment by amolenaar** [comment]
@danyeaw By separating the model and presentation specific properties, I found out that some property editors are now applied to elements they were not applied to before.

E.g. `ActivityParameterNode` now also has _Upper Bound_ and _Ordering_ editors, since it inherits from `ObjectNode`.

image

Now we can explicitly blacklist those uses, but I'm not sure if we should. What do you think?

**Review by danyeaw — COMMENTED** [review]
Hi @amolenaar, wow, this is a really nice feature to be able to show the property editor from the model browser!

One thing I noticed while testing this:
1. Create a Class and a Block
2. Select the Block in the Diagram -> Block Property Page is shown
3. Select the Class in the Model Browser -> Class Property Page is shown
4. Try to click on the Block in the Diagram again -> Class Property Page is still shown

**Comment by amolenaar** [comment]
Thanks for testing this. It's a bit tricky to orchestrate the events coming both from the model browser and diagram. I'll have a look into this.

**Comment by amolenaar** [comment]
@danyeaw I think I resolved the (re)selection issue. `DiagramSelectionChanged` events are now emited explicitly, and not based on changes in the diagram selection.

File: gaphor/UML/actions/activitypropertypage.py [hunk 1]
```diff
@@ -206,47 +207,74 @@ def _on_name_changed(self, entry):
...
+            "show-parameter-direction-editor",
+            signals={
+                "show-direction-changed": (self._on_show_direction_changed,),
+            },
+        )
+
+        show_direction = builder.get_object("show-direction")
+        show_direction.set_active(self.item.show_direction)
+
+        print("(self.item.subject and self.item.subject.parameter)")
```
Comment (danyeaw) [inline gaphor/UML/actions/activitypropertypage.py:276]: One more print statement nit

**Review by danyeaw — APPROVED** [review]
Nice, this is working great now! :star_struck:

File: gaphor/UML/actions/activitypropertypage.py (hunk 1 above)
Comment (amolenaar) [reply gaphor/UML/actions/activitypropertypage.py:276]: Ah no. I need a linting rule that errors on print statements!

**Pull request merged**